databricks/koalas/internal.py

```diff
@@ -454,14 +454,13 @@ def __init__(self, sdf: spark.DataFrame,
         assert isinstance(sdf, spark.DataFrame)
 
         if index_map is None:
-            # Here is when Koalas DataFrame is created directly from Spark DataFrame.
-            assert not any(SPARK_INDEX_NAME_PATTERN.match(name) for name in sdf.schema.names), \
+            assert not any(SPARK_INDEX_NAME_PATTERN.match(name) for name in sdf.columns), \
                 "Index columns should not appear in columns of the Spark DataFrame. Avoid " \
-                "index colum names [%s]." % SPARK_INDEX_NAME_PATTERN
+                "index column names [%s]." % SPARK_INDEX_NAME_PATTERN
```

Comment:
I don't think we still need this check, but I'd leave it as an assertion for now.